aptible/data_source_stack_test.go

```diff
@@ -0,0 +1,75 @@
+package aptible
+
+import (
+	"fmt"
+	"github.com/aptible/go-deploy/aptible"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceStack_validation(t *testing.T) {
+	requiredAttrs := []string{"name"}
+	var testSteps []resource.TestStep
+
+	for _, attr := range requiredAttrs {
+		testSteps = append(testSteps, resource.TestStep{
+			PlanOnly:    true,
+			Config:      `data "aptible_stack" "test" {}`,
+			ExpectError: regexp.MustCompile(fmt.Sprintf("%q is required", attr)),
+		})
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		Steps:             testSteps,
+	})
+}
+
+func TestAccDataSourceStack_deploy(t *testing.T) {
+	var stacks []aptible.Stack
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			client, err := aptible.SetUpClient()
+			if err != nil {
+				t.Fatalf("Unable to generate and setup client for stacks test - %s", err.Error())
+				return
+			}
+
+			stacks, err = client.GetStacks()
+			if err != nil {
+				t.Fatalf("Unable to retrieve stacks for test - %s", err.Error())
+				return
+			}
+			if len(stacks) == 0 {
+				t.Fatal("Unable to find stacks with a zero length")
+				return
+			}
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
```

Comment:
This test doesn't create any resources so it shouldn't `CheckDestroy`.

Reply:
yeah been trying different values to see what i can do to get the test to run in CI without running integration tests...

Reply:
this was one place where i noticed environments were not getting hit, but stacks were missing it so i added it in temporarily. this might be the only place where we don't create a resource and only do a data lookup in a test (data tests didn't exist before)

Reply:
dropping it